lib/spack/spack/database.py

```diff
@@ -65,8 +66,13 @@
     (Version('0.9.3'), Version('5')),
 ]
 
-# Timeout for spack database locks in seconds
-_db_lock_timeout = 120
+# Default timeout for spack database locks in seconds, which we want to provide
+# a fairly quick turnaround for parallel installs
+_db_lock_timeout = 3
+
+# Default timeout for spack package locks in seconds, which we want to provide
+# a very quick turnaround for parallel installs
+_pkg_lock_timeout = None
```

Comment:
low pkg lock timeout makes sense to me, since we now have an algorithm around the package locks, and we can do other things while we wait for them.

Reply:
Looks like an outdated comment that needs to be updated.  

I changed the default to `None` to get the behavior we expect for parallel builds because not all locking operations set/modify the timeout values.

Reply:
Part of the issue here relates to operations that do and do not include a `timeout` argument.  When one is not provided, then the default is used.  When one is provided, it _may_ not be used (e.g., nested locks).